### ISSUE TITLE
systemd-logind does not react to power btn acpi event anymore

### ISSUE THREAD

**Issue by oddfellow** [systemd/systemd#23425]
**systemd version the issue has been seen with**
 > 251~rc3-1-arch

**Used distribution**
 > Arch Linux

**Linux kernel version used** (`uname -a`)

 > `5.17.8-arch1-1`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > Pressing ether physical or virtual power should create a journal entry and power down the system with default config /etc/systemd/logind.conf

**Unexpected behaviour you saw**
 > systemd-logind does not react to the physical or virtual power button event. Nothing is logged to journal.

**Steps to reproduce the problem**
 > Just boot to a tty, no display manager, no gnome, etc. and let systemd-logind handle the power button event: Nothing happens in 251-rc3. With systemd 250.5 the event is logged in journal and the system is powered down.

**Additional program output to the terminal or log subsystem illustrating the issue**

no journal entries with 251-rc3

jounal entries with systemd 250.5
```text
systemd-logind[378]: Power key pressed short.
systemd-logind[378]: Powering Off...
systemd-logind[378]: System is powering down.
```

**Comment by ademasi** [comment]
Same issue here, I had to downgrade to 250.5 too. The "Watching system buttons" lines are not present in the log. Hence, no power button action and laptop lid switch action to shutoff the laptop or suspend.

**Comment by chiraag-nataraj** [comment]
The broken part seems to be in `udev`. On Debian `sid`, I initially downgraded `systemd` (and associated `systemd-*` packages) to `250.4` _without_ touching `udev` (which remained on `251~rc3`). Power Button and Lid Switch events remained broken until I downgraded `udev` as well and rebooted.

**Comment by ademasi** [comment]
@chiraag-nataraj Did you reboot after dowgrading systemd or just restarted logind ?

**Comment by chiraag-nataraj** [comment]
@ademasi I initially tried just running `systemctl restart systemd-logind` and then figured I'd try a reboot. Nothing worked until I downgraded `udev` and then rebooted.

**Comment by yuwata** [comment]
@oddfellow @ademasi @chiraag-nataraj Do you see the logs like the following?
```
May 18 23:15:28 systemd[1]: Starting User Login Management...
May 18 23:15:28 systemd-logind[124905]: New seat seat0.
May 18 23:15:28 systemd-logind[124905]: Watching system buttons on /dev/input/event2 (Power Button)
May 18 23:15:28 systemd-logind[124905]: Watching system buttons on /dev/input/event1 (Lid Switch)
May 18 23:15:28 systemd-logind[124905]: Watching system buttons on /dev/input/event0 (Sleep Button)
May 18 23:15:28 systemd-logind[124905]: Watching system buttons on /dev/input/event3 (AT Translated Set 2 keyboard)
May 18 23:15:28 systemd-logind[124905]: Watching system buttons on /dev/input/event6 (ThinkPad Extra Buttons)
May 18 23:15:28 systemd[1]: Started User Login Management.
```
Then, could you compare the result of e.g. `udevadm info /dev/input/event2` with v250 and v251-rcX?

**Comment by yuwata** [comment]
Also, could you create the following udev rules, reboot the system, and provide the debugging logs of udevd (`journalctl --no-hostname -u systemd-udevd.service`)?
```
# /etc/udev/rules.d/00-debug-input.rules
SUBSYSTEM=="input", OPTIONS="log_level=debug"
```

**Comment by ademasi** [comment]
I reinstall v251-rc3, I made sure to know the path to my power button before (event1), under v250.

```
alex@ ~> journalctl -b -u systemd-logind.service -p6 --no-hostname
May 18 16:41:40 systemd[1]: Starting User Login Management...
May 18 16:41:40 systemd-logind[565]: New seat seat0.
May 18 16:41:40 systemd[1]: Started User Login Management.
May 18 16:41:41 systemd-logind[565]: New session 2 of user alex.
May 18 16:41:41 systemd-logind[565]: New session 1 of user alex.

alex@ ~> udevadm info /dev/input/event1
P: /devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1/event1
M: event1
R: 1
U: input
D: c 13:65
N: input/event1
L: 0
E: DEVPATH=/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1/event1
E: DEVNAME=/dev/input/event1
E: MAJOR=13
E: MINOR=65
E: SUBSYSTEM=input
E: USEC_INITIALIZED=2115247
E: ID_INPUT=1
E: ID_INPUT_KEY=1
E: ID_PATH=acpi-PNP0C0C:00
E: ID_PATH_TAG=acpi-PNP0C0C_00
E: LIBINPUT_DEVICE_GROUP=19/0/1:PNP0C0C/button

```
[debug-input.txt](https://github.com/systemd/systemd/files/8717278/debug-input.txt)

**Comment by yuwata** [comment]
@ademasi Thanks. Could you also show the result of the same commands with v251-rc3?

**Comment by ademasi** [comment]
Please find v250 below:
```

alex@ ~> journalctl -b -u systemd-logind.service -p6 --no-hostname
May 18 16:55:40 systemd[1]: Starting User Login Management...
May 18 16:55:40 systemd-logind[434]: Watching system buttons on /dev/input/event1 (Power Button)
May 18 16:55:40 systemd-logind[434]: Watching system buttons on /dev/input/event0 (Lid Switch)
May 18 16:55:40 systemd-logind[434]: New seat seat0.
May 18 16:55:40 systemd[1]: Started User Login Management.
May 18 16:55:40 systemd-logind[434]: Watching system buttons on /dev/input/event2 (AT Translated Set 2 keyboard)
May 18 16:55:41 systemd-logind[434]: Watching system buttons on /dev/input/event3 (Intel HID events)
May 18 16:55:41 systemd-logind[434]: Watching system buttons on /dev/input/event4 (Intel HID 5 button array)
May 18 16:55:42 systemd-logind[434]: New session 1 of user alex.
May 18 16:55:42 systemd-logind[434]: New session 2 of user alex.
May 18 16:56:06 systemd-logind[434]: Watching system buttons on /dev/input/event16 (Aeropex by AfterShokz (AVRCP))



alex@ ~> udevadm info /dev/input/event1
P: /devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1/event1
N: input/event1
L: 0
E: DEVPATH=/devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input1/event1
E: DEVNAME=/dev/input/event1
E: MAJOR=13
E: MINOR=65
E: SUBSYSTEM=input
E: USEC_INITIALIZED=2006835
E: ID_INPUT=1
E: ID_INPUT_KEY=1
E: ID_PATH=acpi-PNP0C0C:00
E: ID_PATH_TAG=acpi-PNP0C0C_00
E: LIBINPUT_DEVICE_GROUP=19/0/1:PNP0C0C/button
E: TAGS=:power-switch:
E: CURRENT_TAGS=:power-switch:

```

[debug-input.txt](https://github.com/systemd/systemd/files/8717356/debug-input.txt)

**Comment by yuwata** [comment]
Thank you so much. The issue is clearly caused by the missing TAG for the device.

Could you show the result of `udevadm test /dev/input/event1` with v251-rc3?

**Comment by yuwata** [comment]
Ah, no, its our installer issue...

**Comment by yuwata** [comment]
Fix is waiting in #23428. Thank you for all your help!

**Comment by eworm-de** [comment]
Should be fixed in `systemd 251rc3-2`.
